nncf/experimental/onnx/graph/model_transformer.py

```diff
@@ -0,0 +1,108 @@
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+from typing import TypeVar
+
+from copy import deepcopy
+import onnx
+
+from nncf.common.graph.model_transformer import ModelTransformer
+from nncf.common.graph.transformations.commands import TransformationCommand
+from nncf.common.quantization.structs import QuantizationMode
+
+from nncf.experimental.onnx.graph.onnx_graph import ONNXGraph
+
+from nncf.experimental.onnx.graph.transformations.layout import ONNXTransformationLayout
+
+from nncf.experimental.onnx.graph.transformations.commands import ONNXQuantizerInsertionCommand
+
+ModelType = TypeVar('ModelType')
+
+QUANTIZER_NAME_PREFIX = 'QuantizeLinear_'
+DEQUANTIZER_NAME_PREFIX = 'DequantizeLinear_'
+SCALE_TENSOR_NAME_PREFIX = 'scale_'
+ZERO_POINT_NAME_PREFIX = 'zero_point_'
+
+
+# pylint: disable=no-member
+
+class ONNXModelTransformer(ModelTransformer):
+    def __init__(self, model: onnx.ModelProto):
+        super().__init__(model)
+        self.transformed_model = deepcopy(model)
+
+    def transform(self, transformation_layout: ONNXTransformationLayout) -> onnx.ModelProto:
+        for transform in transformation_layout.transformations:
+            self._apply_transformation(transform)
+        return self.transformed_model
+
+    def _apply_transformation(self, transformation: TransformationCommand):
+        if isinstance(transformation, ONNXQuantizerInsertionCommand):
+            self._insert_quantizer_dequantizer(transformation)
+
+    def _insert_quantizer_dequantizer(self, transformation: ONNXQuantizerInsertionCommand):
+        target_point = transformation.target_point
+        scale = transformation.quantizer_parameters.scale
+        zero_point = transformation.quantizer_parameters.zero_point
+        mode = transformation.quantizer_parameters.mode
+
+        per_channel = isinstance(scale, list)
+
+        zero_point = [zero_point] if not isinstance(zero_point, list) else zero_point
+        tensor_type = onnx.TensorProto.UINT8 if mode == QuantizationMode.ASYMMETRIC else onnx.TensorProto.INT8
+        scale = [scale] if not isinstance(scale, list) else scale
+
+        axis = 0 if per_channel else None
+        dims = [len(scale)] if per_channel else []
+
+        quantizer_name = QUANTIZER_NAME_PREFIX + target_point
+        dequantizer_name = DEQUANTIZER_NAME_PREFIX + target_point
+        scale_tensor_name = SCALE_TENSOR_NAME_PREFIX + target_point
+        zero_point_tensor_name = ZERO_POINT_NAME_PREFIX + target_point
+
+        onnx_scale = onnx.helper.make_tensor(scale_tensor_name, onnx.TensorProto.FLOAT, dims, scale)
+        onnx_zero_point = onnx.helper.make_tensor(zero_point_tensor_name, tensor_type, dims, zero_point)
+
+        quantizer = onnx.helper.make_node(
+            'QuantizeLinear',
```

Comment:
Why not to reuse `QUANTIZER_NAME_PREFIX` here?
BTW, QUANTIZER_NAME_PREFIX can be defined without underscore.

Reply:
They have two different meanings: 
1) Node name, which could be any.
2) The standard ONNX node type.
I don't like to use the same constant for them. I would rather use the constants for names.Bbut for the type, I would rather stay with the simple string, because the node name could change, while node type is strict.